internal/appconfig/patches.go

```diff
@@ -37,7 +39,21 @@ func mapToConfig(cfgMap map[string]any) (*Config, error) {
 	if err != nil {
 		return cfg, err
 	}
-	return cfg, json.Unmarshal(newbuf, cfg)
+
+	decoder := json.NewDecoder(bytes.NewReader(newbuf))
+	decoder.DisallowUnknownFields()
+	err = decoder.Decode(cfg)
```

Comment:
can we add a test case and its test file at `internal/appconfig/testdata/` ? 

I wonder if `cfg` is "filled" with the rest of the values even when there is an error due to unknown fields.